pkg/networkpolicy/multinetingressrule.go

```diff
@@ -53,6 +53,40 @@ func (builder *IngressRuleBuilder) WithPortAndProtocol(port uint16, protocol cor
 	return builder
 }
 
+// WithProtocol appends new item with only protocol to Ports list.
+func (builder *IngressRuleBuilder) WithProtocol(protocol corev1.Protocol) *IngressRuleBuilder {
```

Comment:
same comment as other WithProtocol, I think there is an opportunity to validate the protocol